Unity/RuntimeTypeInfo.cs

```diff
@@ -1,31 +1,114 @@
 ﻿using System;
+using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 
 namespace Unity
 {
-    public class RuntimeTypeInfo
+    public unsafe class RuntimeTypeInfo
     {
+        NativeTypeInfoUnion native;

```

Comment:
Like with `NativeObject`, this should mirror the interface approach used for `TypeTree`.

Reply:
I'm having trouble figuring out how to do this. presumably we would change 
```csharp
unsafe struct NativeTypeArray
{
    public int Count;
    public IntPtr First;
}
```
and then create the list of types with 
```csharp
types = new List<RuntimeTypeInfo>(runtimeTypes->Count);
for (int i = 0; i < runtimeTypes->Count; i++)
{
	IntPtr current = GetCurrentFromIndex(runtimeTypes->First, i);
	types.Add(new RuntimeTypeInfo(current, engine.version));
}
```
and then in RuntimeTypeInfo
```csharp
RuntimeTypeInfo(IntPtr ptr, UnityVersion version){
    if (version >= UnityVersion.Unity2017_3) 
    {
        this.native = new V2(ptr);
    } else {
        this.native = new V1(ptr);
    }
}
```
but i'm having a bit of trouble figuring out how to implement `GetCurrentFromIndex`

Reply:
`NativeTypeArray` is a struct containing a flexible array of pointers, so all you should need to do is:
```CSharp
IntPtr current = (&runtimeTypes->First)[i];
types.Add(new RuntimeTypeInfo(current, engine.Version));
```